python/common/org/Python.java

```diff
@@ -395,10 +395,30 @@ public static void initializeModule(java.lang.Class cls, java.util.Map<java.lang
             "As repr(), return a string containing a printable representation of an\n" +
             "object, but escape the non-ASCII characters in the string returned by\n" +
             "repr() using \\x, \\u or \\U escapes.  This generates a string similar\n" +
-            "to that returned by repr() in Python 2.\n"
+            "to that returned by repr() in Python 2.\n",
+        args = {"object"}
     )
-    public static org.python.types.Str ascii() {
-        throw new org.python.exceptions.NotImplementedError("Builtin function 'ascii' not implemented");
+    public static org.python.types.Str ascii(org.python.Object object) {
+        java.lang.String s = ((org.python.types.Str) object.__repr__()).value;
+        StringBuilder resultString = new StringBuilder();
+
+        for(java.lang.Character c : s.toCharArray()){
```

Comment:
Minor style point as our Java code linter isn't up yet: for the `for` and `if` constructs, there should always be a space between the keyword and the left parentheses, and between the right parentheses and left curly brace. Instead of `for(...){` it should be like `for (...) {`.